src/misc/unicode.cpp

```diff
@@ -1361,49 +1885,73 @@ static bool construct_mapping(const uint16_t code_page)
 		}
 	}
 
-	mappings_normalized_by_codepage[code_page] = new_mapping;
-	mappings_reverse_by_codepage[code_page]    = new_mapping_reverse;
+	auto& mappings = per_code_page_mappings[code_page];
+
+	mappings.dos_to_grapheme_normalized = new_mapping;
+	mappings.grapheme_to_dos = new_mapping_reverse;
 
 	// Construct decomposed mapping
-	construct_decomposed(mappings_normalized_by_codepage[code_page],
-	                     mappings_decomposed_by_codepage[code_page]);
+	construct_decomposed(mappings.dos_to_grapheme_normalized,
+	                     mappings.dos_to_grapheme_decomposed);
+
+	// Construct mapping for box-optimized fallback mode
+	construct_box_fallback(mappings.dos_to_grapheme_normalized,
+	                       mappings.box_code_points);
+
+	// Construct upper/lower case mappings
+	construct_case_mapping(uppercase, mappings.dos_to_grapheme_normalized,
+	                       mappings.uppercase);
+	construct_case_mapping(lowercase, mappings.dos_to_grapheme_normalized,
+	                       mappings.lowercase);
 
 	return true;
 }
 
 static void construct_aliases(const uint16_t code_page)
 {
-	assert(!aliases_normalized_by_codepage.count(code_page));
-	assert(!aliases_decomposed_by_codepage.count(code_page));
-	assert(mappings_normalized_by_codepage.count(code_page));
+	auto& mappings = per_code_page_mappings[code_page];
 
-	const auto& mapping      = mappings_normalized_by_codepage[code_page];
-	auto& aliases_normalized = aliases_normalized_by_codepage[code_page];
+	assert(mappings.aliases_normalized.empty());
+	assert(mappings.aliases_decomposed.empty());
 
-	for (const auto& alias : config_aliases) {
-		if (!mapping.count(alias.first) && mapping.count(alias.second) &&
+	const auto& mapping_normalized = mappings.dos_to_grapheme_normalized;
+	auto& aliases_normalized = mappings.aliases_normalized;
+
+	auto add_alias = [&](const alias_t& alias) {
+		if (!mapping_normalized.count(alias.first) &&
+		    mapping_normalized.count(alias.second) &&
 		    !aliases_normalized.count(alias.first)) {
 			aliases_normalized[alias.first] =
-			        mapping.find(alias.second)->second;
+			        mapping_normalized.find(alias.second)->second;
```

Comment:
Suggest quantity comparisons from count.

Reply:
OK, changed - everywhere in this file.